### ISSUE TITLE
ssl_error_no_cypher_overlap when using Posh-ACME cert

### ISSUE THREAD

**Issue by Bagu** [rmbolger/Posh-ACME#250]
Hello,
i'm using the latest Posh-ACME with Cloudflare plugin.
Certificates seem to be ok but, when i'm using it on apache, i get "ssl_error_no_cypher_overlap" and server certificate does NOT include an ID which matches the server name.

Here is the command line to get my certificates :
```
$pArgs = @{ CFTokenInsecure = 'mysecrettoken' }
$pArgs.CFTokenReadAllInsecure = 'myothersecrettoken'
New-PACertificate '*.site1.com','*.site2.fr','*.site3.fr','*.site4.fr' -AcceptTOS -Contact my@email.com -DnsPlugin Cloudflare -PluginArgs $pArgs -Verbose
```

Is there something wrong ? Can you help me ?

**Comment by rmbolger** [comment]
Unfortunately, I don't really have much experience with apache. But the cypher overlap error sounds like an apache misconfiguration to me. Clients and Servers have to agree on the set of encryption protocols and ciphers they support. So if you've tweaked the defaults or are using really old defaults with a really new browser, that sort of error can occur.

The certificate not matching the server name makes it sound like you didn't actually install the certificate you got into apache. Posh-ACME doesn't currently do anything with the certificate once it is retrieved. To add it to your webserver requires additional scripting on your part.

**Comment by Bagu** [comment]
Thank you for your reply.
In fact, I'm using apache's default settings.
The problem only appears when I go through Cloudflare. It seems that the latter is interposed between apache and the web client when it is configured in "proxy" mode
So I decided to leave the universal SSL enabled while leaving cloudflare in "strict" mode
However, I also had another error: "server certificate does NOT include an ID which matches the server name"
I solved this problem by specifying the first level hosts in addition to the wildcards. For example :
`New-PACertificate 'site1.biz', 'site2.fr', 'site3.fr', 'site4.fr', '*. Site1.biz', '*. Site2.fr', '*. Site3.fr' , '*. site4.fr' -AcceptTOS -Contact my@email.fr -DnsPlugin Cloudflare -PluginArgs $ pArgs -Verbose`

I'm not sure it's the best solution, but it works.
